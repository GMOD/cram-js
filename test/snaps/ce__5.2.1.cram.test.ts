import { testFile } from '../testUtil'
testFile('ce#5.2.1.cram')
