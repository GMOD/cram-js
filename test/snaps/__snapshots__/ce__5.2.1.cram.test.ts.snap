// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole ce#5.2.1.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 2,
    "minorVersion": 1,
  },
  {
    "containerHeader": {
      "_endPosition": 39,
      "_size": 13,
      "alignmentSpan": 0,
      "landmarks": [
        0,
      ],
      "length": 10523,
      "numBases": 0,
      "numBlocks": 1,
      "numLandmarks": 1,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 39,
        "_size": 13,
        "alignmentSpan": 0,
        "landmarks": [
          0,
        ],
        "length": 10523,
        "numBases": 0,
        "numBlocks": 1,
        "numLandmarks": 1,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_II",
              },
              {
                "tag": "LN",
                "value": "5000",
              },
              {
                "tag": "M5",
                "value": "8e7993f7a93158587ee897d7287948ec",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_III",
              },
              {
                "tag": "LN",
                "value": "5000",
              },
              {
                "tag": "M5",
                "value": "3adcb065e1cf74fafdbba1e8c352b323",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_IV",
              },
              {
                "tag": "LN",
                "value": "5000",
              },
              {
                "tag": "M5",
                "value": "251af66a69ee589c9f3757340ec2de6f",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_V",
              },
              {
                "tag": "LN",
                "value": "5000",
              },
              {
                "tag": "M5",
                "value": "cf200a65fb754836dcc56b24b3170ee8",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              16,
              2048,
            ],
          },
          "parametersBytes": 7,
        },
        "BS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              0,
              1,
            ],
          },
          "parametersBytes": 6,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "DL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              1,
              100000,
            ],
          },
          "parametersBytes": 8,
        },
        "FC": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              68,
              88,
            ],
          },
          "parametersBytes": 6,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              5,
              6,
            ],
          },
          "parametersBytes": 6,
        },
        "FP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              3,
              3,
              3,
              4,
              5,
              5,
            ],
            "numCodes": 7,
            "numLengths": 7,
            "symbols": [
              6,
              1,
              4,
              5,
              72,
              0,
              26,
            ],
          },
          "parametersBytes": 16,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              100,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              0,
              1,
            ],
          },
          "parametersBytes": 6,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {
        "ASc": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "XGC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "XMC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "XNC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "XOC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "XSc": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
        "YTZ": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 9,
            "stopByte": 9,
          },
          "parametersBytes": 2,
        },
      },
      "tagIdsDictionary": [
        [
          "XGC",
          "XMC",
          "XNC",
          "XOC",
          "XSc",
          "ASc",
          "YTZ",
        ],
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 10579,
      "_size": 17,
      "alignmentSpan": 100100,
      "landmarks": [
        345,
      ],
      "length": 612,
      "numBases": 600,
      "numBlocks": 7,
      "numLandmarks": 1,
      "numRecords": 6,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 2,
    },
    "data": [
      {
        "_endPosition": 10579,
        "_size": 17,
        "alignmentSpan": 100100,
        "landmarks": [
          345,
        ],
        "length": 612,
        "numBases": 600,
        "numBlocks": 7,
        "numLandmarks": 1,
        "numRecords": 6,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 2,
      },
      {
        "features": [
          {
            "alignmentStart": 2,
            "cramFlags": 3,
            "flags": 16,
            "lengthOnRef": 101,
            "mappingQuality": 1,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "X",
                "data": 1,
                "pos": 5,
                "refPos": 6,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 6,
                "refPos": 7,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 12,
                "refPos": 13,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 18,
                "refPos": 19,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 24,
                "refPos": 25,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 28,
                "refPos": 29,
              },
            ],
            "readGroupId": -1,
            "readLength": 100,
            "readName": "I",
            "sequenceId": 0,
            "tags": {
              "AS": -18,
              "XG": 1,
              "XM": 5,
              "XN": 0,
              "XO": 1,
              "XS": -18,
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 10930,
          },
          {
            "alignmentStart": 2,
            "cramFlags": 3,
            "flags": 16,
            "lengthOnRef": 101,
            "mappingQuality": 1,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "X",
                "data": 1,
                "pos": 5,
                "refPos": 6,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 6,
                "refPos": 7,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 12,
                "refPos": 13,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 18,
                "refPos": 19,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 24,
                "refPos": 25,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 28,
                "refPos": 29,
              },
            ],
            "readGroupId": -1,
            "readLength": 100,
            "readName": "II.14978392",
            "sequenceId": 0,
            "tags": {
              "AS": -18,
              "XG": 1,
              "XM": 5,
              "XN": 0,
              "XO": 1,
              "XS": -18,
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 10931,
          },
          {
            "alignmentStart": 2,
            "cramFlags": 3,
            "flags": 16,
            "lengthOnRef": 101,
            "mappingQuality": 1,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "X",
                "data": 1,
                "pos": 5,
                "refPos": 6,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 6,
                "refPos": 7,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 12,
                "refPos": 13,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 18,
                "refPos": 19,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 24,
                "refPos": 25,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 28,
                "refPos": 29,
              },
            ],
            "readGroupId": -1,
            "readLength": 100,
            "readName": "III",
            "sequenceId": 0,
            "tags": {
              "AS": -18,
              "XG": 1,
              "XM": 5,
              "XN": 0,
              "XO": 1,
              "XS": -18,
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 10932,
          },
          {
            "alignmentStart": 2,
            "cramFlags": 3,
            "flags": 16,
            "lengthOnRef": 101,
            "mappingQuality": 1,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "X",
                "data": 1,
                "pos": 5,
                "refPos": 6,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 6,
                "refPos": 7,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 12,
                "refPos": 13,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 18,
                "refPos": 19,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 24,
                "refPos": 25,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 28,
                "refPos": 29,
              },
            ],
            "readGroupId": -1,
            "readLength": 100,
            "readName": "IV",
            "sequenceId": 0,
            "tags": {
              "AS": -18,
              "XG": 1,
              "XM": 5,
              "XN": 0,
              "XO": 1,
              "XS": -18,
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 10933,
          },
          {
            "alignmentStart": 2,
            "cramFlags": 3,
            "flags": 16,
            "lengthOnRef": 101,
            "mappingQuality": 1,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "X",
                "data": 1,
                "pos": 5,
                "refPos": 6,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 6,
                "refPos": 7,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 12,
                "refPos": 13,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 18,
                "refPos": 19,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 24,
                "refPos": 25,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 28,
                "refPos": 29,
              },
            ],
            "readGroupId": -1,
            "readLength": 100,
            "readName": "V",
            "sequenceId": 0,
            "tags": {
              "AS": -18,
              "XG": 1,
              "XM": 5,
              "XN": 0,
              "XO": 1,
              "XS": -18,
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 10934,
          },
          {
            "alignmentStart": 2,
            "cramFlags": 3,
            "flags": 2048,
            "lengthOnRef": 100100,
            "mappingQuality": 1,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "X",
                "data": 0,
                "pos": 1,
                "refPos": 2,
              },
              {
                "code": "X",
                "data": 0,
                "pos": 27,
                "refPos": 28,
              },
              {
                "code": "D",
                "data": 100000,
                "pos": 28,
                "refPos": 29,
              },
              {
                "code": "X",
                "data": 0,
                "pos": 28,
                "refPos": 100029,
              },
              {
                "code": "X",
                "data": 0,
                "pos": 100,
                "refPos": 100101,
              },
            ],
            "readGroupId": -1,
            "readLength": 100,
            "readName": "VI",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10935,
          },
        ],
        "header": {
          "_endPosition": 10963,
          "_size": 34,
          "compressedSize": 34,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            2,
            193,
            135,
            4,
            6,
            0,
            5,
            4,
            1,
            9,
            11,
            12,
            255,
            255,
            255,
            255,
            15,
            141,
            217,
            102,
            124,
            204,
            42,
            226,
            90,
            139,
            251,
            231,
            239,
            221,
            251,
            155,
            86,
          ],
          "contentId": 0,
          "contentPosition": 10929,
          "contentType": "MAPPED_SLICE_HEADER",
          "parsedContent": {
            "_endPosition": 10613,
            "_size": 34,
            "contentIds": [
              1,
              9,
              11,
              12,
            ],
            "md5": [
              141,
              217,
              102,
              124,
              204,
              42,
              226,
              90,
              139,
              251,
              231,
              239,
              221,
              251,
              155,
              86,
            ],
            "numBlocks": 5,
            "numContentIds": 4,
            "numRecords": 6,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 100100,
            "refSeqStart": 2,
          },
          "uncompressedSize": 34,
        },
      },
    ],
  },
]
`;
